app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt

```diff
@@ -36,15 +36,7 @@ class IntentReceiverActivity : Activity() {
     }
 
     suspend fun processIntent(intent: Intent) {
-        if (!Browsers.all(this).isDefaultBrowser) {
-            /* If the user has unset us as the default browser, unset openLinksInAPrivateTab */
-            this.settings().openLinksInAPrivateTab = false
-            components.analytics.metrics.track(Event.PreferenceToggled(
-                preferenceKey = getString(R.string.pref_key_open_links_in_a_private_tab),
-                enabled = false,
-                context = applicationContext
-            ))
-        }
+        DefaultBrowserPreference(applicationContext).unsetOpenLinksInAPrivateTabIfNecessary()
```

Comment:
For later: I realized that we improved the performance of `HomeActivity.onResume` but `IntentReceiverActivity`, used for opening links from external apps and homescreen icons, still has this same performance hit. Can you file a follow-up bug?